src/test/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ClusterMonitorSpec.scala

```diff
@@ -688,30 +697,30 @@ class ClusterMonitorSpec extends TestKit(ActorSystem("leonardotest")) with FlatS
 
     // Create the first cluster
     val supervisor = withClusterSupervisor(gdDAO, computeDAO, iamDAO, storageDAO, authProvider, mockJupyterDAO, false) { actor =>
-      actor ! ClusterCreated(savedCreatingCluster)
-      expectMsgClass(clusterMonitorPatience, classOf[Terminated])
-
-      val updatedCluster = dbFutureValue { _.clusterQuery.getActiveClusterByName(creatingCluster.googleProject, creatingCluster.clusterName) }
-      updatedCluster shouldBe 'defined
-      updatedCluster.map(_.status) shouldBe Some(ClusterStatus.Running)
-      updatedCluster.flatMap(_.dataprocInfo.hostIp) shouldBe Some(IP("1.2.3.4"))
-      updatedCluster.map(_.instances) shouldBe Some(Set(masterInstance, workerInstance1, workerInstance2))
 
+      eventually {
+        val updatedCluster = dbFutureValue {
+          _.clusterQuery.getActiveClusterByName(creatingCluster.googleProject, creatingCluster.clusterName)
+        }
+        updatedCluster shouldBe 'defined
+        updatedCluster.map(_.status) shouldBe Some(ClusterStatus.Running)
+        updatedCluster.flatMap(_.dataprocInfo.hostIp) shouldBe Some(IP("1.2.3.4"))
+        updatedCluster.map(_.instances) shouldBe Some(Set(masterInstance, workerInstance1, workerInstance2))
+      }
       verify(storageDAO, never).deleteBucket(any[GcsBucketName], any[Boolean])
-      // removeIamRolesForUser should not have been called because there is still a creating cluster in the DB
-      verify(iamDAO, never()).removeIamRolesForUser(any[GoogleProject], any[WorkbenchEmail], mockitoEq(Set("roles/dataproc.worker")))
```

Comment:
was this intentionally removed?

Reply:
Yes, since we don't need to send individual messages to the supervisor anymore we only need this check this once.